src/web-ui/src/App.js

```diff
@@ -68,119 +58,119 @@ class App extends Component {
-            this.setState({ people: response.people }, this.getSnapshot);
-          });
-        }
-      }
-    );
-  }
+  };
+
+  const setupWebcam = instance => {
+    setWebcamInstance(instance);
 
-  componentDidMount() {
     const checkIfReady = () => {
-      if (this.webcam && this.webcam.state && this.webcam.state.hasUserMedia) {
-        this.setState({
-          readyToStream: true
-        });
+      if (
+        webcamInstance &&
+        webcamInstance.state &&
+        webcamInstance.state.hasUserMedia
+      ) {
+        setReadyToStream(true);
       } else setTimeout(checkIfReady, 250);
     };
 
     checkIfReady();
-  }
-
-  render() {
-    return (
-      <div className="App">
-        <Header
-          toggleRekognition={this.toggleRekognition}
-          addUser={this.addUser}
-          readyToStream={this.state.readyToStream}
-        />
-        <Grid>
-          <CameraHelp show={!this.state.readyToStream} />
-          <Row>
-            <Col md={8} sm={6}>
-              <Grid>
-                <Row>
-                  <Col md={8} sm={6}>
-                    <Webcam
-                      ref={webcam => (this.webcam = webcam)}
-                      screenshotFormat="image/jpeg"
-                      videoConstraints={{
-                        width: 1280,
-                        height: 640,
-                        facingMode: "user"
-                      }}
-                      width="100%"
-                      height="100%"
-                    />
-                  </Col>
-                  <Col md={4} sm={6}>
-                    <EngagementSummary
-                      detectedFaces={this.state.detectedFaces}
-                      detectedPeople={this.state.detectedPeople}
-                      webcamCoordinates={this.state.webcamCoordinates}
-                    />
-                  </Col>
-                </Row>
-                <Row style={{ marginTop: "20px" }}>
-                  <Col md={4} sm={6}>
-                    <h3>Trends for last hour</h3>
-                    <PolarChart
-                      data={Object.keys(this.state.aggregate).map(
-                        sentiment => ({
-                          x: sentiment,
-                          y: this.state.aggregate[sentiment]
-                        })
-                      )}
-                    />
-                  </Col>
-                  <Col md={4} sm={6}>
-                    <h3 style={{ marginBottom: "40px" }}>Engagement Meter</h3>
-                    <HalfCircleMeter
-                      backgroundColor="#fff"
-                      foregroundColor="#FF9900"
-                      value={this.state.happyometer}
-                      size={250}
-                    />
-                  </Col>
-                </Row>
-              </Grid>
-            </Col>
-          </Row>
-        </Grid>
-      </div>
-    );
-  }
-}
-
-export default App;
+  };
+
+  const toggleRekognition = () => {
```

Comment:
For this functions you can use useCallback, again, not mandatory but good to experiment